src/test/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -122,36 +124,130 @@ public void canMonitorTrip() throws URISyntaxException {
-        simulatedItinerary.legs.get(0).departureDelay = 60 * 20;
+    @ParameterizedTest
+    @MethodSource("createDelayNotificationTestCases")
+    void testDelayNotifications(DelayNotificationTestCase testCase) {
+        TripMonitorNotification notification = testCase.checkMonitoredTrip.checkTripForDelay(testCase.delayType);
+        if (testCase.expectedNotificationMessage == null) {
+            assertNull(notification, testCase.message);
+        } else {
+            assertNotNull(notification, String.format("Expected notification for test case: %s", testCase.message));
+            assertEquals(testCase.expectedNotificationMessage, notification.body, testCase.message);
+        }
+    }
+
+    private static List<DelayNotificationTestCase> createDelayNotificationTestCases () throws URISyntaxException {
+        List<DelayNotificationTestCase> testCases = new ArrayList<>();
+
+        // doesn't create departure/arrival notification for on-time trip
+        CheckMonitoredTrip onTimeTrip = createCheckMonitoredTrip();
+        onTimeTrip.setJourneyState(createDefaultJourneyState());
+        testCases.add(new DelayNotificationTestCase(
+            onTimeTrip,
+            NotificationType.DEPARTURE_DELAY,
+            "doesn't create departure notification for on-time trip"
+        ));
+        testCases.add(new DelayNotificationTestCase(
+            onTimeTrip,
+            NotificationType.ARRIVAL_DELAY,
+            "doesn't create arrival notification for on-time trip"
+        ));
+
+        // creates departure notification for 20 minute late trip
+        CheckMonitoredTrip twentyMinutesLateTimeTrip = createCheckMonitoredTrip();
+        offsetItineraryTime(
+            twentyMinutesLateTimeTrip.matchingItinerary,
+            TimeUnit.MILLISECONDS.convert(20, TimeUnit.MINUTES)
+        );
+        JourneyState twentyMinutesLateJourneyState = createDefaultJourneyState();
+        twentyMinutesLateTimeTrip.setJourneyState(twentyMinutesLateJourneyState);
+        testCases.add(new DelayNotificationTestCase(
+            twentyMinutesLateTimeTrip,
+            NotificationType.DEPARTURE_DELAY,
+            "The departure time for your itinerary is now 09:00 (20 minutes late) (your threshold is currently set to 15 minutes).",
+            "creates departure notification for 20 minute late trip"
+        ));
+        testCases.add(new DelayNotificationTestCase(
+            twentyMinutesLateTimeTrip,
+            NotificationType.ARRIVAL_DELAY,
+            "The arrival time for your itinerary is now 09:18 (20 minutes late) (your threshold is currently set to 15 minutes).",
+            "creates arrival notification for 20 minute late trip"
+        ));
 
+        // doesn't create departure notification for 20 minute late trip w/ 15 minute threshold
+        // doesn't create arrival notification for 20 minute late trip w/ 15 minute threshold
```

Comment:
Am I missing something? Why don't we want to create a notification for a 20 minute late trip when the threshold is 15 minutes? Please enhance comments to help the reader.

Reply:
Added some more comments and changed some timing to note an 18 minute late baseline in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14